<launch>
    <arg name="drone_name" />
    <arg name="repulsion_min_threshold" default="2.0" />
    <arg name="repulsion_max_threshold" default="1.0" />
    <arg name="reduced_vel_threshold" default="4.5" />
    <arg name="halting_threshold" default="3.0" />
    <arg name="restart_threshold" default="3.2" />
    <arg name="repulsion_max_norm" default="0.4" />
    <arg name="reduced_vel_factor" default="0.3" />
    
    <group ns="/collision_avoidance/$(arg drone_name)">
        <node name="collision_avoidance" pkg="drone_coverage" type="drone_collision_avoidance.py" output="screen">
            <param name="drone_name" type="string" value="$(arg drone_name)" />
            <param name="repulsion_min_threshold" type="double" value="$(arg repulsion_min_threshold)" />
            <param name="repulsion_max_threshold" type="double" value="$(arg repulsion_max_threshold)" />
            <param name="reduced_vel_threshold" type="double" value="$(arg reduced_vel_threshold)" />
            <param name="halting_threshold" type="double" value="$(arg halting_threshold)" />
            <param name="restart_threshold" type="double" value="$(arg restart_threshold)" />
            <param name="repulsion_max_norm" type="double" value="$(arg repulsion_max_norm)" />
            <param name="reduced_vel_factor" type="double" value="$(arg reduced_vel_factor)" />
        </node>
    </group>
</launch>